Jenkinsfile.nightly

```diff
@@ -0,0 +1,92 @@
+pipeline {
+    agent { label 'go'}
+    options {
+        buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10')
+        timeout(time: 90, unit: 'MINUTES')
+    }
+    environment {
+        WORKING_DIR = "/home/jenkins/go/src/github.com/kiegroup/kogito-cloud-operator/"
+        GOPATH = "/home/jenkins/go"
+        GOCACHE = "/home/jenkins/go/.cache/go-build"
+    }
+    stages {
+        stage('Clean Workspace') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    deleteDir()
+                }
+            }
+        }
+        stage('Initialize') {
+            steps {
+                sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
+                sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+            }
+        }
+        stage('Build Kogito Operator') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        export GOROOT=`go env GOROOT`
+                        GO111MODULE=on 
+                        go get -u golang.org/x/lint/golint
+                        touch /etc/sub{u,g}id
+                        usermod --add-subuids 10000-75535 \$(whoami)
+                        usermod --add-subgids 10000-75535 \$(whoami)
+                        cat /etc/subuid
+                        cat /etc/subgid
+                        make image_builder=podman
+                    """
+                }
+            }
+            
+        }
+        stage('Build Kogito CLI') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh "make build-cli"
+                }
+            }
+        }
+        stage('Push Operator Image to Openshift Registry') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        set +x && podman login -u jenkins -p \$(oc whoami -t) --tls-verify=false ${OPENSHIFT_REGISTRY}
+                        cd version/ && TAG_OPERATOR=\$(grep -m 1 'Version =' version.go) && TAG_OPERATOR=\$(echo \${TAG_OPERATOR#*=} | tr -d '"')
+                        podman tag quay.io/kiegroup/kogito-cloud-operator:\${TAG_OPERATOR} ${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator:pr-\$(echo \${GIT_COMMIT} | cut -c1-7)
+                        podman push --tls-verify=false docker://${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator:pr-\$(echo \${GIT_COMMIT} | cut -c1-7)
```

Comment:
Maybe you can do `$(echo \${GIT_COMMIT} | cut -c1-7)` using Groovy and set it as an environment variable for better readability?

Reply:
Sure! I totally agree, it will get better. I intend to do that in my next sprint, does it sound good to you? I will take advantage and improve other things in this pipeline.

Reply:
I don't know how you're getting the git commit id, but you can use git to get the short commit id with this command:
git rev-parse --short HEAD

Reply:
git rev-parse --short HEAD is a good option but I need this variable because it is the value of the last commit and it is recognized internally by Jenkins and used by the plugin.